### DISCUSSION TITLE
Database Drivers / and stuff?

### DISCUSSION THREAD

**Discussion by computamike** [openbenches/openbenches.org#345] [General]
Hi @edent  and OB community - I was wondering if you could help me out? 

(I'm not a php guy - I have done stuff with php in the past)
 
I'm working on a docker file for #332, allowing a developer to clone the repo - run a docker-compose command, and get up and running for development.

I have it (i think) mostly working but when I try and set it up, I'm having to add some extra packages. 

I added the recent composer.json file - and performed a `composer install` .

I'm seeing this error in the console : 
![image](https://github.com/openbenches/openbenches.org/assets/464876/33450c86-d24c-4ec8-b903-c9268d5f9065)

so - to get past that I ran a `composer require doctrine/orm` 

And now I get this error - 
![image](https://github.com/openbenches/openbenches.org/assets/464876/5d9718c4-f143-4ec0-90da-8f874f2dd4f5)

which I think is due to the DB Connection string - so I set that in line with the docs  - and I get the same error.  

I think based on the URL - `mysqli:` - but I can't figure out how to install it?  

I'll do more research - but was wondering if anyone had any idea about this?

**Comment by computamike** [comment]
I literally woke up this morning and I thought - Mysqli is going to be a PHP extension - I think all i have to do is configure it to be enabled.

**Comment by pwaring** [comment]
You probably want something like this in your Dockerfile:

RUN docker-php-ext-install mysqli pdo pdo_mysql

That will install and enable the relevant extensions.

**Comment by arizonagroovejet** [comment]
If you haven't already have a look at https://github.com/openbenches/openbenches.org/blob/master/CONTRIBUTING.md which mentions some packages which are needed.

**Reply by computamike** [reply]
thanks - I think i've made some progress... Here is a screen shot of openBenches running on docker desktop - 
![image](https://github.com/openbenches/openbenches.org/assets/464876/6a7f3244-5e41-48f1-b238-cc1f5925a31a)
